File: provable-sdk-go/prove.go
package provable

// ProveData proves data by computing its hash and calling Kayros API
// dataType is optional and defaults to "provable_sdk" padded to 32 bytes
func ProveData(data []byte, dataType ...string) (*ProveSingleHashResponse, error) {
	dataHash := Keccak256(data)
	return ProveSingleHash(dataHash, dataType...)
}

// ProveDataWithOptions proves data by computing its hash and calling Kayros API with explicit request options.
func ProveDataWithOptions(data []byte, opts *RequestOptions) (*ProveSingleHashResponse, error) {
	dataHash := Keccak256(data)
	return ProveSingleHashWithOptions(dataHash, opts)
}

// ProveDataStr proves string data by computing its hash and calling Kayros API
// dataType is optional and defaults to "provable_sdk" padded to 32 bytes
func ProveDataStr(s string, dataType ...string) (*ProveSingleHashResponse, error) {
	dataHash := Keccak256Str(s)
	return ProveSingleHash(dataHash, dataType...)
}

// ProveDataStrWithOptions proves string data by computing its hash and calling Kayros API with explicit request options.
func ProveDataStrWithOptions(s string, opts *RequestOptions) (*ProveSingleHashResponse, error) {
	dataHash := Keccak256Str(s)
	return ProveSingleHashWithOptions(dataHash, opts)
}

File: provable-sdk-rs/src/prove.rs
use crate::api::{prove_single_hash, prove_single_hash_with_options};
use crate::error::Result;
use crate::hash::{hash, hash_str};
use crate::options::RequestOptions;
use crate::types::ProveSingleHashResponse;

pub fn prove_data(
    data: impl AsRef<[u8]>,
    data_type: Option<&str>,
) -> Result<ProveSingleHashResponse> {
    let hashed = hash(data);
    prove_single_hash(&hashed, data_type)
}

pub fn prove_data_with_options(
    data: impl AsRef<[u8]>,
    opts: Option<&RequestOptions>,
) -> Result<ProveSingleHashResponse> {
    let hashed = hash(data);
    prove_single_hash_with_options(&hashed, opts)
}

pub fn prove_data_str(data: &str, data_type: Option<&str>) -> Result<ProveSingleHashResponse> {
    let hashed = hash_str(data);
    prove_single_hash(&hashed, data_type)
}
